### PR TITLE
PEP 622: Write up why we reject alternative spellings of OR patterns

### PR DESCRIPTION
**Pull request by gvanrossum**
Much of this was cribbed from https://github.com/gvanrossum/patma/issues/93

### CODE REVIEW

File: pep-0622.rst [hunk 1]
```diff
@@ -1390,6 +1390,80 @@ similar.  This example is from ``difflib.py`` in the stdlib::
...
+    case 404:
+      print("Some HTTP error")
+
+  This is how this would be done in C, using its fall-through
+  semantics for cases.  However, we don't want to mislead people into
+  thinking that ``match``/``case`` uses fall-through semantics (which
+  are a common source of bugs in C).  Also, this would be a novel
+  indentation pattern, which might make it harder to support in IDEs
+  and such (it would break the simple rule "add an indentation level
+  after a line ending in a colon").
```
Comment (JelleZijlstra) [inline pep-0622.rst:1428]: Also, this alternative wouldn't support nested OR patterns.

File: pep-0622.rst [hunk 2]
```diff
@@ -1390,6 +1390,80 @@ similar.  This example is from ``difflib.py`` in the stdlib::
...
+
+    case 401 or 403 or 404:
+        print("Some HTTP error")
+
+  This could work, and the readability is not too different from using
+  ``|``.  Some users expressed a preference for ``or`` because they
+  associate ``|`` with bitwise OR.  However:
+
+  1. Many other languages that have pattern matching use ``|`` (the
+     list includes Elixir, Erlang, F#, Haskell, Mathematica, OCaml,
```
Comment (JelleZijlstra) [inline pep-0622.rst:1451]: Haskell uses `|` for guards, not OR.

File: pep-0622.rst (hunk 1 above)
Comment (gvanrossum) [reply pep-0622.rst]: Thanks, fixed.

File: pep-0622.rst (hunk 2 above)
Comment (gvanrossum) [reply pep-0622.rst]: Oops, removed.

**Comment by gvanrossum** [comment]
Eh, I'm just going to land this.

**Pull request merged**